src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -808,26 +548,13 @@ void ObjectSynchronizer::notifyall(Handle obj, TRAPS) {
   JavaThread* current = THREAD;
 
   markWord mark = obj->mark();
-  if (LockingMode == LM_LIGHTWEIGHT) {
-    if ((mark.is_fast_locked() && current->lock_stack().contains(obj()))) {
-      // Not inflated so there can't be any waiters to notify.
-      return;
-    }
-  } else if (LockingMode == LM_LEGACY) {
-    if (mark.has_locker() && current->is_lock_owned((address)mark.locker())) {
-      // Not inflated so there can't be any waiters to notify.
-      return;
-    }
+  if ((mark.is_fast_locked() && current->lock_stack().contains(obj()))) {
+    // Not inflated so there can't be any waiters to notify.
+    return;
   }
 
   ObjectMonitor* monitor;
-  if (LockingMode == LM_LIGHTWEIGHT) {
-    monitor = LightweightSynchronizer::inflate_locked_or_imse(obj(), inflate_cause_notify, CHECK);
-  } else {
-    // The ObjectMonitor* can't be async deflated until ownership is
-    // dropped by the calling thread.
-    monitor = inflate(current, obj(), inflate_cause_notify);
-  }
+  monitor = LightweightSynchronizer::inflate_locked_or_imse(obj(), inflate_cause_notify, CHECK);
```

Comment:
same here with
ObjectMonitor* monitor = LIght ...

I think we should have another RFE to look at eliminating the middle call.  Call these in LIghtweightSynchronizer::notify, notifyAll and waitInterruptably directly and remove these functions.